### DISCUSSION TITLE
How to connect built in models with a custom model?

### DISCUSSION THREAD

**Discussion by MStarha** [Relm4/Relm4#488] [Q&A] [answered]
In Relm4 one creates models to represent the state, but some GTK4 widgets have their own models like `SpinButton` or `Scale`. How do I connect the undelying model to my model so that both update simultaneously? Do I have to implement `GObject` functionality for my model and bind the properties?

**Comment by AaronErhardt** [comment, marked as answer]
Relm4 0.6 introduced data bindings to address this problem. However, they are not covered by the book yet.

You can find documentation for them [here](https://docs.rs/relm4/latest/relm4/binding/index.html) and an example [here](https://github.com/Relm4/Relm4/blob/main/examples/data_binding.rs).

I hope this helps you :)

**Reply by MStarha** [reply]
Wonderfull, thank you.
